glyph/src/ttfp.rs

```diff
@@ -19,23 +19,51 @@ impl From<GlyphId> for ttfp::GlyphId {
 
 /// A pre-rendered image of a glyph, usually used for emojis or other glyphs
 /// that can't be represented only using an outline.
+#[deprecated(since = "0.2.21", note = "Deprecated in favor of `GlyphImage2`")]
 #[derive(Debug, Clone)]
 pub struct GlyphImage<'a> {
     /// Offset of the image from the normal origin (top at the baseline plus
     /// ascent), measured in pixels at the image's current scale.
     pub origin: Point,
     /// Current scale of the image in pixels per em.
     pub scale: f32,
-    /// Raw image data (not a bitmap).
+    /// Raw image data, not a bitmap in the case of [`GlyphImageFormat::Png`] format.
     pub data: &'a [u8],
     /// Format of the raw data.
     pub format: GlyphImageFormat,
 }
 
+#[allow(deprecated)]
 impl<'a> From<ttfp::RasterGlyphImage<'a>> for GlyphImage<'a> {
     fn from(img: ttfp::RasterGlyphImage<'a>) -> Self {
+        let img: GlyphImage2 = img.into();
         GlyphImage {
+            origin: img.origin,
+            scale: img.scale,
+            data: img.data,
+            format: img.format,
+        }
+    }
+}
+/// A pre-rendered image of a glyph, usually used for emojis or other glyphs
+/// that can't be represented only using an outline.
+#[non_exhaustive]
+#[derive(Debug, Clone)]
+pub struct GlyphImage2<'a> {
+    pub origin: Point,
+    pub width: usize,
+    pub height: usize,
+    pub scale: f32,
+    pub data: &'a [u8],
+    pub format: GlyphImageFormat,
+}
+
+impl<'a> From<ttfp::RasterGlyphImage<'a>> for GlyphImage2<'a> {
+    fn from(img: ttfp::RasterGlyphImage<'a>) -> Self {
+        Self {
             origin: point(img.x.into(), img.y.into()),
+            width: img.width.into(),
+            height: img.height.into(),
             scale: img.pixels_per_em.into(),
```

Comment:
This is maybe an opportunity to improve this too :thinking: particularly as currently font-scale isn't related to units-per-em. Maybe this should just use the upstream name, I can do this as a follow up change.